python/ray/data/datasource/file_meta_provider.py

```diff
@@ -324,3 +327,25 @@ def prefetch_file_metadata(
             return _fetch_metadata_remotely(pieces)
         else:
             return _fetch_metadata(pieces)
+
+
+def _handle_read_s3_files_error(error: OSError, paths: Union[str, List[str]]) -> str:
+    # Specially handle AWS error when reading files, to give a clearer error message
+    # to avoid confusing users. The real issus is most likely to be AWS S3 file
+    # credential not haven been properly configured yet.
+    if "AWS Error [code 15]: No response body" in str(error):
+        if isinstance(paths, str):
+            # Quote to highlight single file path in error message for better
+            # readability. List of file paths will be shown up as ['foo', 'boo'],
+            # so only quote single file path here.
+            paths = f'"{paths}"'
+        raise PermissionError(
```

Comment:
Don't add `from None` intentionally to still keep the original error message, that still might be useful for debugging in the future, if other unknown errors happen other than credential.